src/main/java/at/spengergasse/cooking/recipes/service/recipe/RecipeService.java

```diff
@@ -0,0 +1,84 @@
+package at.spengergasse.cooking.recipes.service.recipe;
+
+import at.spengergasse.cooking.recipes.domain.Recipe;
+import at.spengergasse.cooking.recipes.domain.CachedUser;
+import at.spengergasse.cooking.recipes.domain.utils.key.Key;
+import at.spengergasse.cooking.recipes.domain.utils.key.KeyType;
+import at.spengergasse.cooking.recipes.persistence.RecipeRepository;
+import at.spengergasse.cooking.recipes.service.image.ImageService;
+import at.spengergasse.cooking.recipes.service.recipe.commands.CreateRecipeCommand;
+import at.spengergasse.cooking.recipes.service.recipe.commands.UpdateLikesCommand;
+import at.spengergasse.cooking.recipes.service.user.PreferenceDto;
+import at.spengergasse.cooking.recipes.service.user.UserDto;
+import at.spengergasse.cooking.recipes.service.user.UserClient;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@AllArgsConstructor
+@Service
+public class RecipeService {
+    private final UserClient userService;
+    private final RecipeRepository recipeRepository;
+
+    // TODO: Proper recipe dto!
+
+    private final ImageService imageService;
+
+    public Recipe createRecipe(CreateRecipeCommand cmd, MultipartFile image){
+        final UserDto user = this.userService.getUser(KeyType.parse(cmd.authorKey()).ensureValid(KeyType.USER));
+
+        String imageUrl = "";
+
+        if(user != null) {
+            if (image != null && !image.isEmpty()) {
+                // Save or process the image as needed
+                // For example, you can save it to a file or a database
+                imageUrl = imageService.uploadImage(image);
+            }
+
+            final Recipe recipe = Recipe.builder()
+                    .title(cmd.title())
+                    .description(cmd.description())
+                    .author(new CachedUser(user))
+                    .ingredients(cmd.ingredientList())
+                    .categories(cmd.categories())
+                    .difficulty(cmd.difficulty())
+                    .titlePictureID(imageUrl)
```

Comment:
strongly believe that you should not store complete urls in the db ... 
just relative references to the bucket store